packages/core/src/api/interactions.ts

```diff
@@ -32,65 +51,166 @@ export class InteractionsAPI {
 	public async reply(
 		interactionId: Snowflake,
 		interactionToken: string,
-		{ files, ...data }: APIInteractionResponseCallbackData & { files?: RawFile[] },
+		body: APIInteractionResponseCallbackData &
+			RESTPostAPIInteractionCallbackQuery & { files?: RawFile[]; with_response?: false },
+		options?: Pick<RequestData, 'signal'>,
+	): Promise<void>;
+
+	/**
+	 * Replies to an interaction
+	 *
+	 * @see {@link https://discord.com/developers/docs/interactions/receiving-and-responding#create-interaction-response}
+	 * @param interactionId - The id of the interaction
+	 * @param interactionToken - The token of the interaction
+	 * @param body - The callback data for replying
+	 * @param options - The options for replying
+	 */
```

Comment:
Everywhere there is an overload for a method, you can remove any documentation on its implementation signature. For example, look at https://github.com/discordjs/discord.js/blob/main/packages/formatters/src/formatters.ts. It's because IntelliSense or the documentation will not pick it up any more.